### PR TITLE
Docs folder + updated docs

### LINKED ISSUE #465: Document logging stuff
I'll create a markdown doc with notes about best practice for logging seeing as it turned out to be more nuanced than we originally expected.

Some details in this comment: https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/issues/349#issuecomment-512650775

@kvakvs Can you comment with some notes on the `Format` method(s) you added to the Log class? Which situations is it best to use those?

I'll do a draft of the doc and send a PR for review.

### LINKED ISSUE #466: Update build instructions docs
Due to #463, we'll need to make sure build instructions (building from source, building form pr, wiki versions of same) include info on the additional setup.

There's several outstanding docs tasks so I'll try and get them all done this weekend.

### PR DESCRIPTION
**Pull request by aubergine10**
- [x] Created new `docs` folder in repo root
- [x] Moved following files in to it:
    * `BUILDING_INSTRUCTIONS.md` - updated content
    * `PR_REVIEW_INSTRUCTIONS.md` - updated content
    * `ATTACHING_DEBUGGER.md` - updated content
- [x] Add logging instructions #465 
- [x] Explain how to flick debug switches (in `LOGGING.md`)
- [x] Refactor `ATTACHING_DEBUGGER.md`

Fixes #466 
Fixes #465

### CODE REVIEW

**Comment by aubergine10** [comment]
@kvakvs How are debug switches turned on and off? Also, if someone wanted to add a new switch in future, is there anything they need to be aware of?

File: docs/BUILDING_INSTRUCTIONS.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+This guide explains how to:
+
+* Install required software
+* Clone the repository (source code) to your local computer
+* Build (compile) the source code to DLL files
+* Test in-game
+
+We strongly recommend doing development work on Windows as there's much more tooling available. If you're using a Mac, set up a Boot Camp partition for Windows and use that. If you're using Linux, consult the internets.
+
+## Install
```
Comment (kvakvs) [inline docs/BUILDING_INSTRUCTIONS.md]: Somewhere around this section you can mention that the solution expects to find `TLM/dependencies` directory with the game C# DLLs copied into it, or a link to `<Steam>/Cities_Skylines/Cities_Data/Managed`.

**Comment by kvakvs** [comment]
@aubergine10 At this moment the only reliable way is to visit affected csproj'ects and add another define to the defines section. Per build configuration. Less global way which only covers 1 file is to add at the top of that file.

File: docs/BUILDING_INSTRUCTIONS.md (hunk 1 above)
Comment (aubergine10) [reply docs/BUILDING_INSTRUCTIONS.md]: The reference path is mentioned but I didn't know about the `TLM/depedencies` folder, will add that in.

Comment (kvakvs) [reply docs/BUILDING_INSTRUCTIONS.md]: If you symlink it before trying the first build, it just finds everything and builds and never requires any references manipulation.

**Comment by aubergine10** [comment]
Not sure I understand the comment above. Is that for adding a complete new switch?

If I wanted to, for example, enable logging for parking AI, which already exists, how would that be achieved?

**Comment by kvakvs** [comment]
Oh do you mean the debug switches in the XML file? I never done that, and it probably could use some usability improvements.

**Comment by krzychu124** [comment]
You have to add (`TLM/State/ConfigData/DebugSettings.cs`):
- default value to `Switches` array,
- new `DebugSwitch` enum value,
- new boolean value in section of `TMPE_GlobalConfig.xml` file before running game
```xml 
<Debug>
    <Switches> 
        <boolean>false</boolean>
         ...
    </Switches>
</Debug>
```
To turn off/on features just change correct <boolean> value from `false` to `true` (index is the same as in `DebugSettings` Switches array) and run game. 
That should work while in main game menu too but I didn't tested that (change value in xml, save file and then in mod options `Maintenance -> Reload global configuration`)

File: docs/BUILDING_INSTRUCTIONS.md (hunk 1 above)
Comment (aubergine10) [reply docs/BUILDING_INSTRUCTIONS.md]: @kvakvs I've updated build docs to include note on the `TLM\dependencies` folder

**Comment by aubergine10** [comment]
Urg, this is a nasty way to do the debug switch toggles in the xml file:

```xml
  <Debug>
    <Switches>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
      <boolean>false</boolean>
    </Switches>
    <ExtVehicleType>None</ExtVehicleType>
    <ExtPathMode>None</ExtPathMode>
  </Debug>
```

Is there any way we could add some sort of human readable hint to those elements? Even if it was just an incrementing number eg. `<boolean switch="1"> ...`

**Comment by aubergine10** [comment]
This is ready for final review @krzychu124 @kvakvs

**Comment by aubergine10** [comment]
Yeah, they work reliably as is so I think leave them for now.

**Pull request merged**